pkg/i2gw/providers/kong/converter.go

```diff
@@ -51,25 +51,25 @@ func (c *converter) convert(storage *storage) (i2gw.GatewayResources, field.Erro
 		ingressList = append(ingressList, *ingress)
 	}
 
-	globalErrs := field.ErrorList{}
-
 	// Convert plain ingress resources to gateway resources, ignoring all
 	// provider-specific features.
 	gatewayResources, errs := common.ToGateway(ingressList, c.implementationSpecificOptions)
 	if len(errs) > 0 {
-		globalErrs = append(globalErrs, errs...)
+		return i2gw.GatewayResources{}, errs
```

Comment:
Returning the error early here is a better approach in my opinion, as otherwise we get cascading error which are hard to understand.
Fox example, this was before:
```
# Encountered 4 errors # paths.backends[0].service.port: Invalid value: "name": named ports not supported: http # <nil>: Internal error: HTTPRoute does not exist # <nil>: Internal error: HTTPRoute does not exist # <nil>: Internal error: HTTPRoute does not exist
```

While this is after:
```
# Encountered 1 errors # paths.backends[0].service.port: Invalid value: "name": named ports not supported: http
```

Reply:
Good point. I think it might be better to do it after `crds.TCPIngressToGatewayAPI` though. 
The reason is, that if you fail here, and then there is a problem reading in `TCPIngressToGatewayAPI` then you'd need the second run to know about it. However if you aggregate the errs, the use can fix them both in the first iteration.

Similarly, I think it is useful to aggregate the featureParse errors for the same reason.

Reply:
So my proposal is to aggregate `toGateway` and `TCPIngressToGatewayAPI`, then `MergeGateways` - leave as it is now, and lastly aggregate the featureParser errors, also kinda as it is now - maybe only `globalErrs` can have a better name

Reply:
sounds good to me